include/envoy/http/codec.h

```diff
@@ -260,6 +283,16 @@ class ClientConnection : public virtual Connection {
    * @return StreamEncoder& supplies the encoder to write the request into.
    */
   virtual StreamEncoder& newStream(StreamDecoder& response_decoder) PURE;
+
+  /**
+   * Called when the connection goes over its high watermark.
```

Comment:
Though long I think it would be more clear if this was called `onBackingConnectionAboveWriteBufferHighWatermark()` (with updated comment per above). Similar below. Thoughts?